pkg/operator/ceph/nfs/controller.go

```diff
@@ -187,6 +190,13 @@ func (r *ReconcileCephNFS) reconcile(request reconcile.Request) (reconcile.Resul
 	// The CR was just created, initializing status fields
 	if cephNFS.Status == nil {
 		updateStatus(k8sutil.ObservedGenerationNotAvailable, r.client, request.NamespacedName, k8sutil.EmptyStatus)
```

Comment:
Needs to return an error so that Rook can be sure the uninitialized status is getting set. That needs to be present to ensure the next status update after key creation works correctly